### ISSUE TITLE
DXVK 2.5.2 causes FPS drop due to shader issues

### ISSUE THREAD

**Issue by RouatbiH** [doitsujin/dxvk#4537]
FPS go to 5 due to shader issue
I tested builds from actions and I found that this become happening after merging #4398

### System information
- GPU: NVIDIA GeForce GTX 1650
- Driver: 32.0.15.6636
- DXVK version: 2.5.2

### Apitrace file(s)
[logs + videos](https://drive.google.com/drive/folders/1hrbNvCZVigGu9whdRWkPLE4WTW5pmcvl?usp=sharing)

**Comment by K0bin** [comment]
Could you please also tell us the name of the game? This is highly specific to one single game, we've tested dozens of games and didn't encounter this problem once.

**Comment by WinterSnowfall** [comment]
Will definitely need to know what game this is.

How is the performance if you run it with `DXVK_LOG_LEVEL=none`?

**Comment by RouatbiH** [comment]
> Could you please also tell us the name of the game? This is highly specific to one single game, we've tested dozens of games and didn't encounter this problem once.

The game has been discontinued for years, and it is no longer officially playable. However, I am running it on a local private server that I developed myself.

Would you prefer that I close this issue for now? I can open a new one if I encounter a similar issue with another game in the future.

**Comment by RouatbiH** [comment]
> Will definitely need to know what game this is.
> 
> How is the performance if you run it with `DXVK_LOG_LEVEL=none`?

Still the same performance

**Comment by WinterSnowfall** [comment]
> Would you prefer that I close this issue for now? I can open a new one if I encounter a similar issue with another game in the future.

No, it's a genuine issue apparently, so we do want to look into it. @Blisto91 has found the game, so we'll see what we can do. It seems to hit the same validation that is needed by The Void, but it's affecting it negatively.

The character model for the robot doesn't seem to be rendered any longer, so it's not just a performance issue.

**Comment by RouatbiH** [comment]
> > Would you prefer that I close this issue for now? I can open a new one if I encounter a similar issue with another game in the future.
> 
> No, it's a genuine issue apparently, so we do want to look into it. @Blisto91 has found the game, so we'll see what we can do. It seems to hit the same validation that is needed by The Void, but it's affecting it negatively.
> 
> The character model for the robot doesn't seem to be rendered any longer, so it's not just a performance issue.

Thank you for the update. I appreciate you looking into it. If you need any additional information or testing on my end, feel free to reach out. You can contact me on Discord: **razrusheniye**.

**Comment by Blisto91** [comment]
> No, it's a genuine issue apparently, so we do want to look into it. @Blisto91 has found the game, so we'll see what we can do.

Scratch that. I had only found a website for the game. Didn't know it was discontinued

**Comment by WinterSnowfall** [comment]
@RouatbiH Please try this build: https://github.com/doitsujin/dxvk/actions/runs/12453032682/artifacts/2353129099 and let us know how it goes. The issue should be sorted now.

**Comment by RouatbiH** [comment]
> @RouatbiH Please try this build: https://github.com/doitsujin/dxvk/actions/runs/12453032682/artifacts/2353129099 and let us know how it goes. The issue should be sorted now.

Thank you for the build! I tested it, and the issue is resolved. I really appreciate your help.

**Comment by Blisto91** [comment]
@RouatbiH Could i get you to dump the shaders of the game? We'd like to take a look at the offending ones that are causing trouble.
Just launch the game with `DXVK_SHADER_DUMP_PATH=` pointing to a empty folder you've created and do the usual thing with the game for this issue. Then archive the dumped shaders and attach them here.

**Comment by WinterSnowfall** [comment]
> @RouatbiH Please try this build: https://github.com/doitsujin/dxvk/actions/runs/12453032682/artifacts/2353129099

P.S.: You need to do that using this build or dxvk 2.5.1, otherwise the relevant shaders will be skipped.

**Comment by RouatbiH** [comment]
> @RouatbiH Could i get you to dump the shaders of the game? We'd like to take a look at the offending ones that are causing trouble. Just launch the game with `DXVK_SHADER_DUMP_PATH=` pointing to a empty folder you've created and do the usual thing with the game for this issue. Then archive the dumped shaders and attach them here.

Here is the link to the archived shaders:
[shaders.zip](https://drive.google.com/file/d/188-jyDOQl1Y5yOgU4FzExdYY5ad5iJ4f/view?usp=sharing)
> P.S.: You need to do that using this build or dxvk 2.5.1, otherwise the relevant shaders will be skipped.

The shaders were dumped using the build you provided (commit 86aa8ba).

**Comment by WinterSnowfall** [comment]
@RouatbiH Thank you for your patience and your support in narrowing down the problem. I am hoping we have a definitive fix for the issue now. Please try this build with the game: https://github.com/doitsujin/dxvk/actions/runs/12654635033/artifacts/2396592897 and let us know if it solves the issue you were seeing.

**Comment by RouatbiH** [comment]
Thank you for the updated build! I tested it, and it's working perfectly now. I really appreciate your effort and the great work on this project. Keep up the fantastic work!
